### ISSUE TITLE
CUDA 7.0 RC Compile Errors

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/picongpu#630]
### Warning

```
src/libPMacc/include/mappings/simulation/EnvironmentController.hpp(70):
  warning: extended friend syntax is a C++11 feature
```

### Errors

```
src/libPMacc/include/cuSTL/container/CartBuffer.tpp(84,113,126mo):
  error: no instance of overloaded function "PMacc::container::CartBuffer<Type, _dim, Allocator, Copier, Assigner>::CartBuffer" matches the specified type
```

### Environment

Hypnos with
```
Currently Loaded Modulefiles:
  1) mpfr/3.0.0             3) gmp/4.3.1              5) cmake/3.0.1            7) boost/1.54.0           9) pngwriter/0.5.4       11) libsplash/1.2.3
  2) mpc/0.8.2              4) gcc/4.6.2              6) openmpi/1.6.3          8) cuda/7.0.ea           10) hdf5-parallel/1.8.14
```

**Comment by ax3l** [comment]
@Heikman can you have a look, too pls? It is caused by cuSTL's `container::CartBuffer`.

**Comment by psychocoderHPC** [comment]
I am working on this BUG. 
I have fixed all compile time errors and running now in runtime errors.

**Comment by f-schmitt-zih** [comment]
Did you move to 7.0 RC already?

**Comment by ax3l** [comment]
oh yes, let us update the issue title since the problem persits.

**Comment by psychocoderHPC** [comment]
solved with #748

**Comment by psychocoderHPC** [comment]
With the fix #748 we get new warnings #749.
#749 is solved with #750
